src/Features/CSharp/Portable/Completion/KeywordRecommenders/ReadOnlyKeywordRecommender.cs

```diff
@@ -26,8 +27,11 @@ public ReadOnlyKeywordRecommender()
 
         protected override bool IsValidContext(int position, CSharpSyntaxContext context, CancellationToken cancellationToken)
         {
+            var previousToken = context.LeftToken.GetPreviousTokenIfTouchingWord(position);
+
             return
                 context.IsGlobalStatementContext ||
+                (previousToken.IsKind(SyntaxKind.RefKeyword) && previousToken.Parent.IsKind(SyntaxKind.Parameter, SyntaxKind.RefType)) ||
```

Comment:
don't mix complex && and || expressions.  just break this nested check out into it's own method (or local).